### PR TITLE
expression: implement vectorized evaluation for `builtinTrim2ArgsSig`

### PR DESCRIPTION
**Pull request by shihongzhi**
### What problem does this PR solve? 

implement vectorized evaluation for builtinRightSig , for #12106

### What is changed and how it works?

    	$go test -v -benchmem -bench=BenchmarkVectorizedBuiltinStringFunc -run=BenchmarkVectorizedBuiltinStringFunc -args "builtinTrim2ArgsSig"
	goos: darwin
	goarch: amd64
	pkg: github.com/pingcap/tidb/expression
	BenchmarkVectorizedBuiltinStringFunc/builtinTrim2ArgsSig-VecBuiltinFunc-4         	   39956	     31806 ns/op	       0 B/op	       0 allocs/op
	BenchmarkVectorizedBuiltinStringFunc/builtinTrim2ArgsSig-NonVecBuiltinFunc-4      	   23016	     68560 ns/op	       0 B/op	       0 allocs/op
	PASS
	ok  	github.com/pingcap/tidb/expression	4.836s

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by qw4990** [comment]
```
[2019-10-09T16:19:03.051Z] FAIL: builtin_string_vec_test.go:96: testEvaluatorSuite.TestVectorizedBuiltinStringEvalOneVec
[2019-10-09T16:19:03.051Z] 
[2019-10-09T16:19:03.051Z] builtin_string_vec_test.go:97:
[2019-10-09T16:19:03.051Z]     testVectorizedEvalOneVec(c, vecBuiltinStringCases)
[2019-10-09T16:19:03.051Z] bench_test.go:601:
[2019-10-09T16:19:03.051Z]     c.Assert(c1.GetString(i), Equals, c2.GetString(i), commentf(i))
[2019-10-09T16:19:03.051Z] ... obtained string = ""
[2019-10-09T16:19:03.051Z] ... expected string = "4vkdHnSLJZ9z"
[2019-10-09T16:19:03.051Z] ... case {retEvalType:3 childrenTypes:[3 3] childrenFieldTypes:[] geners:[]}, row: 1, rowData: [KindBytes 4vkdHnSLJZ9z KindBytes btET86nAdao]
[2019-10-09T16:19:03.051Z] 
[2019-10-09T16:19:03.051Z] 
[2019-10-09T16:19:03.051Z] ----------------------------------------------------------------------
[2019-10-09T16:19:03.051Z] FAIL: builtin_string_vec_test.go:100: testEvaluatorSuite.TestVectorizedBuiltinStringFunc
[2019-10-09T16:19:03.051Z] 
[2019-10-09T16:19:03.051Z] builtin_string_vec_test.go:101:
[2019-10-09T16:19:03.051Z]     testVectorizedBuiltinFunc(c, vecBuiltinStringCases)
[2019-10-09T16:19:03.051Z] bench_test.go:838:
[2019-10-09T16:19:03.051Z]     c.Assert(val, Equals, output.GetString(i), commentf(i))
[2019-10-09T16:19:03.051Z] ... obtained string = "0aRmPetWXhEv3az047"
[2019-10-09T16:19:03.051Z] ... expected string = ""
[2019-10-09T16:19:03.051Z] ... func: builtinTrim2ArgsSig, case {retEvalType:3 childrenTypes:[3 3] childrenFieldTypes:[] geners:[]}, row: 1, rowData: [KindBytes 0aRmPetWXhEv3az047 KindBytes MDj5WetKQd3]
```
@shihongzhi  Please solve the  CI problem.

File: expression/builtin_string_vec_test.go [hunk 1]
```diff
@@ -34,7 +34,9 @@ var vecBuiltinStringCases = map[string][]vecExprBenchCase{
 	ast.Unhex: {
 		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETString}, geners: []dataGenerator{&randHexStrGener{10, 100}}},
 	},
-	ast.Trim:      {},
+	ast.Trim: {
+		{retEvalType: types.ETString, childrenTypes: []types.EvalType{types.ETString, types.ETString}},
```
Comment (XuHuaiyu) [inline expression/builtin_string_vec_test.go]: We may need to specify the data generators to test the cases when `len(str) < len(remstr)` or `len(str) >= len(remstr)`.

Comment (shihongzhi) [reply expression/builtin_string_vec_test.go]: I use randLenStrGener to generate different length case

**Pull request merged**